libsolidity/ast/Types.cpp

```diff
@@ -2698,6 +2708,113 @@ Type const* TupleType::mobileType() const
 	return TypeProvider::tuple(move(mobiles));
 }
 
+BoolResult InlineArrayType::isImplicitlyConvertibleTo(Type const& _other) const
+{
+	auto arrayType = dynamic_cast<ArrayType const*>(&_other);
+
+	if (!arrayType || arrayType->isByteArrayOrString())
+		return false;
+	else
+	{
+		if (!arrayType->isDynamicallySized() && arrayType->length() != components().size())
+			return false; //TODO return string
+
+		for (Type const* c: components())
+			if (!c->isImplicitlyConvertibleTo(*arrayType->baseType()))
```

Comment:
Here we should also return an explanatory string that tells which component fails to convert.